### PR TITLE
KOGITO-3622 - Add grafana to kogito-infra

### PR DESCRIPTION
**Pull request by r00ta**
Jira: https://issues.redhat.com/browse/KOGITO-3622

Scenario: Trusty Monitoring
As you can see in the following schema 
overall (1)1
there are two external resources: prometheus and grafana. The prometheus instance is already managed by the operator, but the grafana resources not. 

The workflow is the following: 
1) A kogito runtime application with the monitoring addon is compiled and a docker image is built. That application will expose the name of the dashboards under the path `/monitoring/dashboards/list.json`. `list.json` contains the name of the dashboards that can be downloaded at `/monitoring/dashboards/xxx.json`. 
2) In case the user uses the grafana `kogito-infra` resource, he has to create it. Otherwise he has to install and deploy a grafana operator resource. 
3) When the kogito runtime application is deployed with the operator (a special flag `enableGrafanaDashboards` must be set) ,then the operator will fetch the dashboards and will create grafana dashboards out of them -> the related jira ticket is the following:   https://issues.redhat.com/browse/KOGITO-3624

Please, note that the final feature is composed by the following PR/ticket and the ticket https://issues.redhat.com/browse/KOGITO-3624 that still has to be implemented by me. 

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors' guide](https://github.com/kiegroup/kogito-cloud-operator/blob/master/README.md#contributing-to-the-kogito-operator)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: pkg/controller/kogitoinfra/grafana/resource.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	} else if !exits {
+		log.Debug("Kogito grafana instance is not exists")
+		return nil, nil
+	} else {
+		log.Debug("Kogito grafana instance found")
+		return grafanaInstance, nil
+	}
+}
+
+func createNewGrafanaInstance(cli *client.Client, name string, namespace string, instance *v1alpha1.KogitoInfra, scheme *runtime.Scheme) (*grafana.Grafana, error) {
```
Comment (spolti) [inline pkg/controller/kogitoinfra/grafana/resource.go:51]: will we going continue to provision infra for user?

File: pkg/controller/kogitoinfra/kogitoinfra_service.go [hunk 2]
```diff
@@ -55,3 +59,8 @@ func IsKafkaResource(instance *v1alpha1.KogitoInfra) bool {
 func isKeycloakResource(instance *v1alpha1.KogitoInfra) bool {
 	return instance.Spec.Resource.APIVersion == infrastructure.KeycloakAPIVersion && instance.Spec.Resource.Kind == infrastructure.KeycloakKind
 }
+
+// isGrafanaResource check if provided KogitoInfra instance is for Keycloak resource
```
Comment (spolti) [inline pkg/controller/kogitoinfra/kogitoinfra_service.go:63]: grafana?

**Review by spolti — CHANGES_REQUESTED** [review]
just a little thing.

@ricardozanini haven't we decided to no create infra anymore or will continue doing it and later let this only for dev mode?

**Review by ricardozanini — CHANGES_REQUESTED** [review]
Hi @r00ta! 

Thanks for your PR, but this is not the strategy we are aiming for 1.0

`KogitoInfra` is meant to **bind** the Kogito services with third party infra, not creating the infra itself. We just kept Kafka/Infinispan/Keycloak for now for backward compatibility.

For Kafka: We create KafkaTopics on already created Kafka instance.
For Infinispan: We fetch Infinispan server details and inject it into the service, so it could connect to it
For Knative Eventing: We bind a given broker with a Kogito service via SinkBindings and Triggers.
For Prometheus: We fetch for the `metrics` endpoint and create `ServiceMonitors` to bind then together, no `KogitoInfra` needed.

For Graphana: Users should create their own Graphana instance and the operator would fetch for the `/dashboards` endpoint, if it's in there, we create [GraphanaDashboards](https://github.com/integr8ly/grafana-operator/blob/master/documentation/dashboards.md) to be discovered by the Graphana Operator. We **won't create** Graphana instances whatsoever.

I'd say that we won't merge this PR since it's not the strategy we are aiming.

**Comment by r00ta** [comment]
Hi @ricardozanini , thanks for the clarification. np, I'll implement what you just suggested. I'll close this

**Comment by ricardozanini** [comment]
> Hi @ricardozanini , thanks for the clarification. np, I'll implement what you just suggested. I'll close this

Take a look at #602 it's pretty similar: we fetch for topics and create Knative resources.

**Pull request closed without merging**